### PR TITLE
login method and findByEmail method added

### PR DESCRIPTION
**Pull request by ira-levch**

### CODE REVIEW

File: src/main/java/mate/academy/service/UserService.java [hunk 1]
```diff
@@ -15,6 +15,11 @@ public class UserService {
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
+        for (User tempUser: users) {
```
Comment (okuzan) [inline src/main/java/mate/academy/service/UserService.java:18]:
```suggestion
        for (User user : users) {
```
use space before and after semicolon

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 2]
```diff
@@ -11,6 +13,14 @@ public class AuthenticationService {
...
      */
     public boolean login(String email, String password) {
+        UserService serv = new UserService();
+        User user = serv.findByEmail(email);
+        if (user != null) {
+            if (user.getPassword().equals(password)) {
+                return true;
+            }
+        }
+        System.out.println("No user with this e-mail is found");
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:23]: don't forget to remove sout (system.out.println) from your solution before committing your HW

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 3]
```diff
@@ -11,6 +13,14 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
+        UserService serv = new UserService();
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:16]: don't shorten your local variables' names, be verbose! 🙏  Remember, developers spend much more time reading the code of others than writing their own, time saved by not typing out a few extra letters is not worth it

Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:16]:
```suggestion
        UserService userService = new UserService();
```

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 4]
```diff
@@ -11,6 +13,14 @@ public class AuthenticationService {
...
+        User user = serv.findByEmail(email);
+        if (user != null) {
+            if (user.getPassword().equals(password)) {
+                return true;
+            }
+        }
+        System.out.println("No user with this e-mail is found");
         return false;
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:24]: try to rewrite this solution without using boolean literals (true/false) at all. Use logical expressions directly in return

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job, some comments 👇

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package mate.academy.service;
 
+import mate.academy.model.User;
+
 public class AuthenticationService {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/AuthenticationService.java:5]:
```suggestion
public class AuthenticationService {
    private final UserService userService = new UserService();
```

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 6]
```diff
@@ -11,6 +13,13 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
+        UserService newService = new UserService();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/AuthenticationService.java:16]:
```suggestion
```

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 7]
```diff
@@ -11,6 +13,13 @@ public class AuthenticationService {
...
+        User user = newService.findByEmail(email);
+        if (user != null) {
+            if (user.getPassword().equals(password)) {
+                return true;
+            }
+        }
         return false;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/AuthenticationService.java:23]:
```suggestion
        return user != null && user.getPassword().equals(password));
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/AuthenticationService.java (hunk 5 above)
Comment (ira-levch) [reply src/main/java/mate/academy/service/AuthenticationService.java:5]: I recceive the message: "Modifier 'private' is not allowed here"

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 8]
```diff
@@ -11,6 +13,8 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
-        return false;
+        final UserService newService = new UserService();
+        User user = newService.findByEmail(email);
+        return user != null && user.getPassword().equals(password));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/AuthenticationService.java:18]:
```suggestion
        return user != null && user.getPassword().equals(password);
```

File: src/main/java/mate/academy/service/AuthenticationService.java (hunk 5 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/service/AuthenticationService.java:5]: There is no any error messages
Screenshot 2024-12-10 at 17 51 17

File: src/main/java/mate/academy/service/UserService.java [hunk 9]
```diff
@@ -15,6 +15,11 @@ public class UserService {
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
+        for (User temporalUser : users) {
+            if (temporalUser.getEmail().equals(email)) {
+                return temporalUser;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/UserService.java:20]:
```suggestion
        for (User user : users) {
            if (user.getEmail().equals(email)) {
                return user;
```

**Pull request closed without merging**